core/src/main/scala/akka/kafka/javadsl/Consumer.scala

```diff
@@ -92,7 +92,12 @@ object Consumer {
     def drainAndShutdown(ec: Executor): CompletionStage[T] =
       control.drainAndShutdown(streamCompletion, ec)
 
-    override def isShutdown: CompletionStage[Done] = control.isShutdown
+    override def isShutdown: CompletionStage[Done] = _isShutdown
+
+    private[this] lazy val _isShutdown =
```

Comment:
Not totally sure if this is needed for Java API bincompat or if we can make `isShutdown` itself lazy...

Reply:
I think you should skip the lazy and just make it a val on both sides, a single future or a future + completion stage alloc per DrainingControl does not seem like a problem.